lib/web/integrations_awsoidc.go

```diff
@@ -329,3 +329,40 @@ func (h *Handler) awsOIDCListEC2ICE(w http.ResponseWriter, r *http.Request, p ht
 		EC2ICEs:   resp.EC2ICEs,
 	}, nil
 }
+
+// awsOIDCDeployC2ICE creates an EC2 Instance Connect Endpoint.
+func (h *Handler) awsOIDCDeployEC2ICE(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (any, error) {
+	ctx := r.Context()
+
+	var req ui.AWSOIDCDeployEC2ICERequest
+	if err := httplib.ReadJSON(r, &req); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	awsClientReq, err := h.awsOIDCClientRequest(r.Context(), req.Region, p, sctx, site)
```

Comment:
You keep alternating between `ctx` and `r.Context()` even though they are the same. I would pick one and be consistent.

Reply:
Replaced all usages of `r.Context` :+1: